### PR TITLE
Swift: Improve documentation and examples that use encryption

### PR DESCRIPTION
**Pull request by geoffw0**
Swift: Improve query documentation and examples that involve encryption:
- replace pseudocode in examples with more realistic code (for the `swift/cleartext-*` queries).
- mention key management in the `.qhelp`s, and provide a reference to help with that.
- use the correct initializer for [CryptoSwift's `CBC` class](https://github.com/krzyzanowskim/CryptoSwift/blob/main/Sources/CryptoSwift/BlockMode/CBC.swift#L31) in examples and tests.

### CODE REVIEW

File: swift/ql/src/queries/Security/CWE-311/CleartextStorageDatabase.qhelp [hunk 1]
```diff
@@ -12,7 +12,7 @@
 </recommendation>
 <example>
 
-<p>The following example shows three cases of storing information using the Core Data library. In the 'BAD' case, the data that is stored is sensitive (a credit card number) and is not encrypted.  In the 'GOOD' cases, the data is either not sensitive, or is protected with encryption.</p>
+<p>The following example shows three cases of storing information using the Core Data library. In the 'BAD' case, the data that is stored is sensitive (a credit card number) and is not encrypted. In the 'GOOD' cases, the data is either not sensitive, or is protected with encryption. When encryption is used, care must be taken to select a secure modern encryption algorithm and put suitable key management practices into place.</p>
```
Comment (subatoi) [inline swift/ql/src/queries/Security/CWE-311/CleartextStorageDatabase.qhelp:15]:
```suggestion
<p>The following example shows three cases of storing information using the Core Data library. In the 'BAD' case, the data that is stored is sensitive (a credit card number) and is not encrypted. In the 'GOOD' cases, the data is either not sensitive, or is protected with encryption. When encryption is used, take care to select a secure modern encryption algorithm, and put suitable key management practices into place.</p>
```

Just a minor comment to make the sentence active (I won't duplicate it across the other instances where I'd make the same recommendation :) )

**Review by subatoi — APPROVED** [review]
Thank you! Just the "one" comment

File: swift/ql/src/queries/Security/CWE-311/CleartextTransmission.qhelp [hunk 2]
```diff
@@ -12,7 +12,7 @@
 </recommendation>
 <example>
 
-<p>The following example shows three cases of transmitting information. In the 'BAD' case, the data transmitted is sensitive (a credit card number) and is not encrypted.  In the 'GOOD' cases, the data is either not sensitive, or is protected with encryption.</p>
+<p>The following example shows three cases of transmitting information. In the 'BAD' case, the data transmitted is sensitive (a credit card number) and is not encrypted.  In the 'GOOD' cases, the data is either not sensitive, or is protected with encryption. When encryption is used, care must be taken to select a secure modern encryption algorithm and put suitable key management practices into place.</p>
```
Comment (jketema) [inline swift/ql/src/queries/Security/CWE-311/CleartextTransmission.qhelp:15]: Probably want to make the same change here as @subatoi suggests above.

File: swift/ql/src/queries/Security/CWE-312/CleartextStoragePreferences.qhelp [hunk 3]
```diff
@@ -12,7 +12,7 @@
 </recommendation>
 <example>
 
-<p>The following example shows three cases of storing information using UserDefaults. In the 'BAD' case, the data that is stored is sensitive (a credit card number) and is not encrypted.  In the 'GOOD' cases, the data is either not sensitive, or is protected with encryption.</p>
+<p>The following example shows three cases of storing information using UserDefaults. In the 'BAD' case, the data that is stored is sensitive (a credit card number) and is not encrypted.  In the 'GOOD' cases, the data is either not sensitive, or is protected with encryption. When encryption is used, care must be taken to select a secure modern encryption algorithm and put suitable key management practices into place.</p>
```
Comment (jketema) [inline swift/ql/src/queries/Security/CWE-312/CleartextStoragePreferences.qhelp:15]: Same.

File: swift/ql/src/queries/Security/CWE-311/CleartextTransmission.qhelp (hunk 2 above)
Comment (geoffw0) [inline swift/ql/src/queries/Security/CWE-311/CleartextTransmission.qhelp:15]: i.e.
```suggestion
<p>The following example shows three cases of transmitting information. In the 'BAD' case, the data transmitted is sensitive (a credit card number) and is not encrypted.  In the 'GOOD' cases, the data is either not sensitive, or is protected with encryption. When encryption is used, take care to select a secure modern encryption algorithm and put suitable key management practices into place.</p>
```

File: swift/ql/src/queries/Security/CWE-312/CleartextStoragePreferences.qhelp (hunk 3 above)
Comment (geoffw0) [reply swift/ql/src/queries/Security/CWE-312/CleartextStoragePreferences.qhelp:15]: Yep.
```suggestion
<p>The following example shows three cases of storing information using UserDefaults. In the 'BAD' case, the data that is stored is sensitive (a credit card number) and is not encrypted.  In the 'GOOD' cases, the data is either not sensitive, or is protected with encryption. When encryption is used, take care to select a secure modern encryption algorithm and put suitable key management practices into place.</p>
```

File: swift/ql/src/queries/Security/CWE-311/CleartextStorageDatabase.qhelp (hunk 1 above)
Comment (geoffw0) [reply swift/ql/src/queries/Security/CWE-311/CleartextStorageDatabase.qhelp]: Thanks, I've made the change in all three places.

**Review by jketema — APPROVED** [review]

**Pull request merged**